crates/trie/sparse/src/trie.rs

```diff
@@ -964,29 +964,12 @@ impl SparseTrieInterface for SerialSparseTrie {
         self.rlp_buf.clear();
     }
 
+    #[inline]
```

Comment:
sorry, I meant adding an `#[inline]` attribute on `check_value_match`, and that's basically the only change that's needed in this PR, because we already don't clone anything until we return the error